swagger.yaml

```diff
@@ -1,42 +1,244 @@
+    required:
+      - "id"
+      - "last_updated"
+      - "links"
+      - "search_index_name"
+      - "state"
+    properties:
+      id:
+        type: string
+      last_updated:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      links:
+        type: object
+        required:
+          - "tasks"
+          - "self"
+        properties:
+          tasks:
+            type: string
+            format: http://localhost:<port>/jobs/{id}/tasks
+          self:
+            type: string
+            format: http://localhost:<port>/jobs/{id}
+      number_of_tasks:
+        type: integer
+      reindex_completed:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      reindex_failed:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      reindex_started:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      search_index_name:
+        type: string
+      state:
+        type: string
+        enum:
+          - CREATED
+          - IN-PROGRESS
+          - COMPLETED
+          - FAILED
+      total_search_documents:
+        type: integer
+
+  NewJob:
     type: object
+    required:
+      - "last_updated"
+      - "search_index_name"
+      - "state"
     properties:
-      message:
+      last_updated:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      number_of_tasks:
+        type: integer
+      reindex_completed:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      reindex_failed:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      reindex_started:
         type: string
-        description: "Message returned by hello world endpoint"
-        example: "Hello, world!"
\ No newline at end of file
+        format: date
+        description: ISO8601 timestamp
+      search_index_name:
+        type: string
+      state:
+        type: string
+        enum:
+          - CREATED
+          - IN-PROGRESS
+          - COMPLETED
+          - FAILED
+      total_search_documents:
+        type: integer
+
+  Jobs:
+    type: object
+    required:
+      - "count"
+      - "items"
+      - "limit"
+      - "offset"
+      - "total_count"
+    properties:
+      count:
+        type: integer
+      items:
+        type: array
+        items:
+          $ref: '#/definitions/Job'
+      limit:
+        type: integer
+      offset:
+        type: integer
+      total_count:
+        type: integer
+      sort:
+        type: integer
+
+parameters:
+
+  id:
+    name: id
+    description: The id of a search reindex job when job POSTed (Job.id)
+    in: path
+    required: true
+    type: string
+
+  limit:
+    name: limit
+    description: The maximum number of Job objects we're returning in this Jobs object. The default limit is 20.
+    in: query
+    type: integer
+    default: 20
```

Comment:
required : true  to be added to limit and offset as they are needed when requesting list of jobs. Do we need sort and state to be added to the query parameter? if so they need this field applied as well

Reply:
limit, offset, sort and state are all optional fields; so they should **not** be set to required. Values offset, limit and sort will have defaults.

state query parameter will need to handle a list of comma separated values and are restricted to a list of enums. This parameter is also not required as the API uses this parameter to filter by state(s)

Reply:
I'm not sure if I've done this correctly. The Reindex API spec shows * next to each of the fields in the 'Jobs' resource, except for the sort field, and so I understood this to mean that those fields are required. However, I've made the swagger now show them as optional.

Reply:
So these fields are optional in the request (query parameters) but required/mandatory in the response body.